homeassistant/components/media_player/yamaha.py

```diff
@@ -0,0 +1,99 @@
+"""
+Support for Yamaha Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.yamaha/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice)
+from homeassistant.const import STATE_OFF, STATE_ON
+REQUIREMENTS = ['rxv==0.1.9']
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_YAMAHA = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yamaha platform."""
+    yamaha = YamahaDevice()
+    if yamaha.update():
+        add_devices([yamaha])
+        return True
+    else:
+        return False
+
+
+class YamahaDevice(MediaPlayerDevice):
+    """Representation of a Yamaha device."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    def __init__(self):
+        """Initialize the Yamaha Receiver."""
+        import rxv
+        self._receiver = rxv.find()[0]
+        self.update()
+        self._name = "Yamaha Receiver"
+        self._host = ""
+        if self._receiver.on:
+            self._pwstate = STATE_ON
+        else:
+            self._pwstate = STATE_OFF
+        self._muted = False
+        self._volume = (self._receiver.volume/100) + 1
+
+    def update(self):
+        """Get the latest details from the device."""
+        self._volume = (self._receiver.volume/100) + 1
+        if self._receiver.on:
+            self._pwstate = STATE_ON
+        else:
+            self._pwstate = STATE_OFF
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._pwstate
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self._volume
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_YAMAHA
+
+    def turn_off(self):
+        """Turn off media player."""
+        self._receiver.on = False
+
+    def set_volume_level(self, volume):
+        """Set volume level, range 0..1."""
+        base_ten = 100-(volume * 100)
+        negative_base_ten = -base_ten
```

Comment:
why not simplify this to `(volume * 100) - 100` or `(volume - 1) * 100`, to more closely match the `(receiver_volume/100)+1` in the other direction.